### PR TITLE
feat(schema): allow add groupsName to not generate automatically the groups name

### LINKED ISSUE #2249: Specify the OpenAPI schema name when using the @Groups decorator
## Information

When the `@Groups` decorator is used in a controller, the OpenAPI Specification (OAS) will create schemas with names specific to the used groups.
For example, given this controller
```ts
@Controller('/news')
class NewsController {
    async create(
        @BodyParams()
        @Groups('action.create', 'scope.admin')
        @Required()
        news: News,
    ) {
        // ...
    }
}
```

the OAS will contain a schema named `NewsActionCreateScopeAdmin`.

When we change the groups, the schema will be renamed accordingly. Since our clients are generated from the OAS, a change in the groups will create new types for the clients as well.

To prevent this from happening, we would like to be able to specify the name of the schema instead of having it be generated based on the groups.

You've made a suggestion of possible solutions:
> Maybe something like `@Groups(...).GroupsName()`. or alternative `@Groups("Name", ['groups'])` by introducing another signature

Either solution would be fine.

## Acceptance criteria

When it is possible to specify the name of a schema when using the `@Groups` decorator.

### PR DESCRIPTION
**Pull request by Romakita**
Closes: #2249

### CODE REVIEW

File: docs/docs/model.md [hunk 1]
```diff
@@ -731,7 +731,65 @@ You can combine different group labels or use a glob pattern to match multiple g
...
+
+In this example, the Groups annotation `@Groups("creation", "summary") user: User` will generate a new model name `UserCreationSummary`.
+If you change the groups list by this one:
+
+```
+ @Groups("creation", "summary", "extra") user: User
+```
+
+The new model name will be `UserCreationSummaryExtra`. This change will break the entire consumer code by removing the `UserCreationSummary` type and giving a new `UserCreationSummaryExtra` type.
+In fact, `UserCreationSummaryExtra` and `UserCreationSummaryExtra` are the same model with more fields!
```
Comment (michael-pxr) [inline docs/docs/model.md]: You probably meant `UserCreationSummary` and `UserCreationSummaryExtra`:
```suggestion
In fact, `UserCreationSummary` and `UserCreationSummaryExtra` are the same model with more fields!
```

File: docs/docs/model.md [hunk 2]
```diff
@@ -731,7 +731,65 @@ You can combine different group labels or use a glob pattern to match multiple g
 to use negation by prefixing the group label with `!`.
 :::
 
-## Groups class definition <Badge text="6.48.0+" />
+## Groups Name
+
+By default, Groups decorator generate automatically a name for each model impacted by the given groups list.
+If you use a typed client http generator based on Swagger (OAS3) to generate the client code, this behavior can be a constraint for your consumer when change the group list.
```
Comment (michael-pxr) [inline docs/docs/model.md]:
```suggestion
If you use a typed client http generator based on Swagger (OAS3) to generate the client code, this behavior can be a constraint for your consumer when you change the group list.
```

**Comment by Romakita** [comment]
@michael-pxr thanks for the review ;)

**Pull request merged**